scene/main/viewport.cpp

```diff
@@ -2295,6 +2279,20 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 	}
 
 	if (mm.is_null() && mb.is_null() && p_event->is_action_type()) {
+		const bool resize_allowed = !DisplayServer::get_singleton()->window_get_flag(DisplayServer::WINDOW_FLAG_RESIZE_DISABLED);
+
+		if (p_event->is_pressed() && p_event->is_action_pressed("toggle_fullscreen") && resize_allowed) {
```

Comment:
Perhaps the check `&& resize_allowed` is superfluous? Some people turn off window resizing but allow fullscreen mode.
And if someone doesn’t need either, then they can simply delete all events from the `toogle_fullscreen` action.

Reply:
> Some people turn off window resizing but allow fullscreen mode.

That doesn't really make sense to me. If your window is set to be non-resizable, it's usually because your game isn't designed for multiple resolutions. Otherwise, there's no reason not to allow it :slightly_smiling_face:

Reply:
> If your window is set to be non-resizable, it's usually because your game isn't designed for multiple resolutions.

Absolutely true, but the game can still work in fullscreen. The resolution does not change, the screen simply stretches (`viewport` mode). This is usually for pixel games.

```
[display]

window/size/width=320
window/size/height=180
window/size/resizable=false
window/size/test_width=640
window/size/test_height=360
window/stretch/mode="viewport"
window/stretch/aspect="keep"
```